sim-lib/src/cln.rs

```diff
@@ -101,10 +107,44 @@ impl LightningNode for ClnNode {
 
     async fn track_payment(
         &mut self,
-        _hash: PaymentHash,
-        _shutdown: Listener,
+        hash: PaymentHash,
+        shutdown: Listener,
     ) -> Result<PaymentResult, LightningError> {
-        unimplemented!()
+        loop {
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => {
+                    break Err(LightningError::TrackPaymentError("Shutdown before tracking results".to_string()));
+                },
+                response = self
+                .client
+                .list_pays(ListpaysRequest {
+                    payment_hash: Some(hash.0.to_vec()),
+                    ..Default::default()
+                }) => {
+                    let ListpaysResponse { pays } = response.map_err(|err| LightningError::TrackPaymentError(err.to_string()))?.into_inner();
+                    if let Some(pay) = pays.first() {
+                        let payment_status = ListpaysPaysStatus::from_i32(pay.status)
+                            .ok_or(LightningError::TrackPaymentError("Invalid payment status".to_string()))?;
+
+                        let payment_outcome = match payment_status {
+                            ListpaysPaysStatus::Pending => continue,
+                            ListpaysPaysStatus::Complete => PaymentOutcome::Success,
+                            // Task: https://github.com/bitcoin-dev-project/sim-ln/issues/26#issuecomment-1691780018
+                            ListpaysPaysStatus::Failed => PaymentOutcome::UnexpectedError,
+                        };
+
+                        break Ok(PaymentResult {
+                            // Task: https://github.com/bitcoin-dev-project/sim-ln/issues/26#issuecomment-1691780018
+                            htlc_count: 1,
+                            payment_outcome,
+                        });
+                    }
+                }
+            }
+
+            time::sleep(Duration::from_millis(100)).await;
```

Comment:
Can we also select on `sleep` + `shutdown` here? That lets us exit a little faster if we shut down while tracking the payment. 

If we do this we can probably bump it up to polling once a second so that we don't hammer CLN too much.

Reply:
select kind of fun

Reply:
Aren't we sleeping twice now?

Feels like this sleep should be removed and the duration of the sleep in `select` should be reduced.

Reply:
ack. I was inattentive!